src/Impostor.Server/Net/Inner/Objects/InnerPlayerControl.cs

```diff
@@ -420,6 +428,30 @@ internal void Die(DeathReason reason)
             PlayerInfo.LastDeathReason = reason;
         }
 
+        internal void Protect(InnerPlayerControl guardianAngel)
+        {
+            // NOTE: Vanilla dispells all GA shields when a kill is blocked, so it suffices to keep track of the last protection action
+            ProtectedOn = _dateTimeProvider.UtcNow;
+            ProtectedBy = guardianAngel;
+        }
+
+        internal bool IsProtected()
+        {
+            // HnS doesn't have guardian angels
+            if (Game.Options.GameMode == GameModes.Normal && ProtectedOn != null)
+            {
+                var opts = (NormalGameOptions)Game.Options;
+                var guardianAngelOpts = (GuardianAngelRoleOptions)opts.RoleOptions.Roles[RoleTypes.GuardianAngel].RoleOptions;
+                var duration = guardianAngelOpts.ProtectionDurationSeconds;
+                var protectionExpiresAt = ProtectedOn.Value.AddSeconds(duration);
+                return protectionExpiresAt >= _dateTimeProvider.UtcNow;
+            }
+            else
+            {
+                return false;
+            }
+        }
```

Comment:
1. 🍝
2. Parameter-less method starting with `Is` sounds wrong, shouldn't this be a property?

Reply:
1. I welcome improvements to this method. It's verbose due to how the options api works
2. We could turn this into a field called `Protected` but eh